v1/charm-dev.yaml

```diff
@@ -0,0 +1,99 @@
+description: A development and testing environment for charmers
+version: latest
+
+instances:
+  charm-dev:
+    image: 20.04
+    limits:
+      min-cpu: 2
+      min-mem: 4G
+      min-disk: 30G
+    timeout: 900
+    cloud-init:
+      vendor-data: |
+        package_update: true
+
+        packages:
+        - python3-pip
+        - jq
+        - gh
+        - sysstat
+        - zsh
+        - fzf
+        - tox
+
+        snap:
+          commands:
+          - snap install --classic juju
+          - snap install --classic microk8s
+          - snap alias microk8s.kubectl kubectl
+          - snap install --classic charmcraft
+          - snap install yq
+          - snap refresh
+
+        runcmd:
+        - DEBIAN_FRONTEND=noninteractive apt -y upgrade
+
+        - |
+          # disable swap
+          sysctl -w vm.swappiness=0
+          echo "vm.swappiness = 0" | tee -a /etc/sysctl.conf
+          swapoff -a
+
+        - |
+          # disable unnecessary services
+          systemctl disable man-db.timer man-db.service --now
+          systemctl disable apport.service apport-autoreport.service  --now
+          systemctl disable apt-daily.service apt-daily.timer --now
+          systemctl disable apt-daily-upgrade.service apt-daily-upgrade.timer --now
+          systemctl disable unattended-upgrades.service --now
+          systemctl disable motd-news.service motd-news.timer --now
+          systemctl disable bluetooth.target --now
+          systemctl disable ua-messaging.service ua-messaging.timer --now
+          systemctl disable ua-timer.timer ua-timer.service --now
+          systemctl disable systemd-tmpfiles-clean.timer --now
+
+        - |
+          # various tools via apt
+          # gh cli
+          curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo gpg --dearmor -o /usr/share/keyrings/githubcli-archive-keyring.gpg
+          echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+
+        - |
+          # apt cleanup
+          apt-get autoremove -y
+
+        - |
+          # oh-my-zsh + juju plugin
+          sudo -u ubuntu sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)" "" --unattended
+          sudo -u ubuntu git clone https://github.com/zsh-users/zsh-autosuggestions.git ~ubuntu/.oh-my-zsh/custom/plugins/zsh-autosuggestions
+          sudo -u ubuntu git clone https://github.com/zsh-users/zsh-syntax-highlighting.git ~ubuntu/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
+          sudo -u ubuntu sed -i 's/plugins=(git)/plugins=(fzf git zsh-autosuggestions zsh-syntax-highlighting virtualenv colored-man-pages juju colorize)/g' ~ubuntu/.zshrc
+          sudo -u ubuntu sed -i 's/ZSH_THEME="robbyrussell"/ZSH_THEME="bureau"/g' ~ubuntu/.zshrc
```

Comment:
Should we change the `ubuntu` user's shell, then?

Reply:
I'd prefer for it to use the default bash shell tbh.

Reply:
And if that's the case, does it make sense to do these ZSH-specific steps? And even if so, is changing the theme the right thing to do?